services/frontend-service/pkg/interceptors/interceptors.go

```diff
@@ -133,13 +149,27 @@ func DexLoginInterceptor(
 	httpHandler http.HandlerFunc,
 	clientID, baseURL string,
 ) {
-	role, err := auth.VerifyToken(req.Context(), req, clientID, baseURL)
+	claims, err := auth.VerifyToken(req.Context(), req, clientID, baseURL)
 	if err != nil {
 		// If user is not authenticated redirect to the login page.
 		http.Redirect(w, req, auth.LoginPATH, http.StatusFound)
 	}
-	auth.WriteUserRoleToHttpHeader(req, role)
-	httpCtx := auth.WriteUserRoleToGrpcContext(req.Context(), role)
+	var httpCtx context.Context
+
+	switch {
```

Comment:
Straightup copy:

```
sub is the subject of the JWT– the user who requested the token, typically an email address. By using it, we can send and receive a small, unique piece of data about a user along with each request, without any additional requests.
```

https://jakeworth.com/til-what-is-the-jwt-sub/

So I think it might be the mail or also a token so it is probably needed to identify the user and important to be passed to the cd-service

Reply:
But then it seems that nothing from the screenshot you provided in your last story can be used for assigning the group for getting the role?

Reply:
Another remark I just noted (cause my Dex Setup failed because of it):
You might need to catch the error of the cast `([]interface{})`. I got a panic e.g. at when "groups" is not present in the claim.

Reply:
Just for reference the stack-trace - The errror was the following:
```log
2024/04/29 19:59:46 http: panic serving 127.0.0.6:42125: interface conversion: interface {} is []string, not []interface {}
goroutine 495 [running]:
net/http.(*conn).serve.func1()
    /usr/local/go/src/net/http/server.go:1868 +0xb9
panic({0xd30a20?, 0xc00023b2c0?})
    /usr/local/go/src/runtime/panic.go:920 +0x270
github.com/freiheit-com/kuberpult/pkg/auth.VerifyToken({0x100b920, 0xc00023a7b0}, 0x1e000000001?, {0xc000046708, 0xd}, {0xc00004a067, 0x2f})
    /kp/pkg/auth/dex.go:300 +0x48e
github.com/freiheit-com/kuberpult/services/frontend-service/pkg/interceptors.DexLoginInterceptor({0x1009f38, 0xc00059c380}, 0xc000323900, 0xc000305288, {0xc000046708, 0xd}, {0xc00004a067, 0x2f})
    /kp/services/frontend-service/pkg/interceptors/interceptors.go:137 +0xa5
github.com/freiheit-com/kuberpult/services/frontend-service/pkg/cmd.runServer.func3({0x1009f38?, 0xc00059c380?}, 0x78aa08d9e478?)
    /kp/services/frontend-service/pkg/cmd/server.go:331 +0x126
net/http.HandlerFunc.ServeHTTP(0xc000067400?, {0x1009f38?, 0xc00059c380?}, 0xc000679520?)
    /usr/local/go/src/net/http/server.go:2136 +0x29
net/http.(*ServeMux).ServeHTTP(0xdd1a40?, {0x1009f38, 0xc00059c380}, 0xc000323900)
    /usr/local/go/src/net/http/server.go:2514 +0x142
github.com/freiheit-com/kuberpult/services/frontend-service/pkg/cmd.runServer.func5({0x1009f38, 0xc00059c380}, 0xc000323900)
    /kp/services/frontend-service/pkg/cmd/server.go:429 +0x98c
net/http.HandlerFunc.ServeHTTP(0x100b920?, {0x1009f38?, 0xc00059c380?}, 0xb?)
    /usr/local/go/src/net/http/server.go:2136 +0x29
github.com/freiheit-com/kuberpult/services/frontend-service/pkg/cmd.(*Auth).ServeHTTP.func1({0x100b920?, 0xc00023a600?})
    /kp/services/frontend-service/pkg/cmd/server.go:538 +0x579
github.com/freiheit-com/kuberpult/pkg/logger.Wrap({0x100b958, 0xc00060a1e0}, 0xc000679990)
    /kp/pkg/logger/logger.go:77 +0x283
github.com/freiheit-com/kuberpult/services/frontend-service/pkg/cmd.(*Auth).ServeHTTP(0xc00023a570?, {0x1009f38?, 0xc00059c380?}, 0x78a9c22087c0?)
    /kp/services/frontend-service/pkg/cmd/server.go:512 +0x8c
github.com/freiheit-com/kuberpult/pkg/setup.(*CORSMiddleware).ServeHTTP(0xc0000110f8, {0x1009f38, 0xc00059c380}, 0xc000322d00)
    /kp/pkg/setup/cors.go:44 +0x933
net/http.(*ServeMux).ServeHTTP(0x169d960?, {0x1009f38, 0xc00059c380}, 0xc000322d00)
    /usr/local/go/src/net/http/server.go:2514 +0x142
net/http.serverHandler.ServeHTTP({0xc000384720?}, {0x1009f38?, 0xc00059c380?}, 0x6?)
    /usr/local/go/src/net/http/server.go:2938 +0x8e
net/http.(*conn).serve(0xc000098b40, {0x100b920, 0xc000528120})
    /usr/local/go/src/net/http/server.go:2009 +0x5f4
created by net/http.(*Server).Serve in goroutine 12
    /usr/local/go/src/net/http/server.go:3086 +0x5cb

```

Reply:
Already removed that line here